test/format/backup.c

```diff
@@ -348,6 +372,104 @@ copy_file(WT_SESSION *session, const char *name)
  */
 #define HOME_BACKUP_INIT_CMD "rm -rf %s/BACKUP %s/BACKUP.copy && mkdir %s/BACKUP %s/BACKUP.copy"
 
+/*
+ * restore_backup_info --
+ *     If it exists, restore the backup information. Return 0 on success.
+ */
+static int
+restore_backup_info(WT_SESSION *session, ACTIVE_FILES *active)
+{
+    FILE *fp;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    size_t len;
+    uint64_t id;
+    uint32_t i;
+    char buf[512], *path;
+
+    testutil_assert(g.c_backup_incr_flag == INCREMENTAL_BLOCK);
+    len = strlen(g.home) + strlen(BACKUP_INFO_FILE) + 2;
+    path = dmalloc(len);
+    testutil_check(__wt_snprintf(path, len, "%s/%s", g.home, BACKUP_INFO_FILE));
+    if ((fp = fopen(path, "r")) == NULL && errno != ENOENT)
+        testutil_die(errno, "restore_backup_info fopen: %s", path);
+    free(path);
+    if (errno == ENOENT)
+        return (1);
```

Comment:
`errno` is not set on success, and it's possible that `errno` has a residual value of `ENOENT` before the call to `fopen`.  So probably need something like `if (fp == NULL && errno == ENOENT)` here, or zero `errno` earlier.